proxy/tracing.go

```diff
@@ -112,26 +121,33 @@ func (t *proxyTracing) startFilterTracing(operation string, ctx *context) *filte
 	if t.disableFilterSpans {
 		return nil
 	}
-	span := tracing.CreateSpan(operation, ctx.request.Context(), t.tracer)
+
+	_, span := t.Start(ctx.request.Context(), operation)
 	ctx.parentSpan = span
 
 	return &filterTracing{span, t.logFilterLifecycleEvents}
 }
 
 func (t *filterTracing) finish() {
-	if t != nil {
-		t.span.Finish()
```

Comment:
why not let the `!= nil` and replace the block with `t.span.End()` ?

Reply:
This is because this function went through a few changes, and before I had two different behaviors here because the logic to decide between Otel and Ot was leaking to this package. But now there is no reason for that, its no problem to have the block inside the `!= nil`.